shared-libs/cht-datasource/src/remote/libs/remote-environment.ts

```diff
@@ -0,0 +1,9 @@
+export const fetchIdentifiedResource = async <T>(resourcePath: string, identifier: string): Promise<Nullable<T>> => {
+  const response = await fetch(`${resourcePath}/${identifier}`);
```

Comment:
What's our nodejs support for CHT 4.x situation again? I couldn't find any info in our docs
I see that fetch was shipped with node v18

Reply:
We are rolling [Node 20](https://github.com/medic/cht-core/blob/900218e9b4594c11e01809d1f1333cd955e7d310/api/Dockerfile#L6)!!!!!

Reply:
Last time I checked fetch isn't API locked until Node 22 so I've been holding off.

Does `resourcePath` include the host and credentials? If so this will have issues with basic auth and you should use the pouchdb instance which has the cookie auth plugin to fetch remotely instead.

Reply:
> Last time I checked fetch isn't API locked until Node 22 so I've been holding off.

Ahh, good to know.  (FWIW, looks like it actually became stable [in Node 21](https://github.com/nodejs/node/blob/main/doc/changelogs/CHANGELOG_V21.md#21.0.0), but still not helpful for us at this moment.)

> Does resourcePath include the host and credentials? If so this will have issues with basic auth and you should use the pouchdb instance which has the cookie auth plugin to fetch remotely instead.

@garethbowen the idea of the `remote` adapter here was to be able to leverage our REST apis and not require Pouch _on the client side_.  (See my [data flow diagram](https://github.com/medic/cht-core/issues/9065#issuecomment-2090866133) for how this fits together in my head...)  Honestly I have been kicking the authentication questions down the road till now because I was not sure how the specifics should work here.  FWIW, when I originally smoke tested making `fetch` calls from webapp code, I found that I could just run something like `fetch('api/v1/person/my-uuid')` and it would somehow magically make the request with the correct root url and credentials (I assume it was using the session cookie somehow).  Of course, things will be different for sentinel/3rd-party apps....

I assume, for the HTTP requests to the API, that we can use `request-promise-native` instead of `fetch`, but that will not solve the authentication problem.  :thinking:   I plan to do more digging here, but def welcome any thoughts/suggestions!

Reply:
Sorry you're absolutely right - ignore my auth comment. Requests sent from the browser will get the cookie auth automatically.

I think I was getting ahead of myself thinking about how this could also be a wrapper tool for Node based scripts because it'll be easier to deal with than pure REST (ie: Oktokit to GitHubs REST API) but this is way out of scope for today.

Reply:
So, I ended up having to back out the `request-promise-native` code. :disappointed: It was causing some janky errors when getting transitively included in webapp. E.g.:

```
../node_modules/assert-plus/assert.js:5:13-37 - Error: Module not found: Error: Can't resolve 'stream' in '/var/home/jlkuester7/git/cht-core/node_modules/assert-plus'

BREAKING CHANGE: webpack < 5 used to include polyfills for node.js core modules by default.
This is no longer the case. Verify if you need this module and configure a polyfill for it.

If you want to include a polyfill, you need to:
        - add a fallback 'resolve.fallback: { "stream": require.resolve("stream-browserify") }'
        - install 'stream-browserify'
If you don't want to include a polyfill, you can use an empty module like this:
        resolve.fallback: { "stream": false }
```

The best solutions I could come up with were:

- Inject a HTTP "service" into cht-datasource so that api/sentinel could use rpn, but webapp could inject the Angular HTTP service.
- Pollyfill the node stuff into webapp so we don't get these errors.
- Just switch back to the `fetch` api and run the risk that something changes between 20 and 21

I have gone with the last option for now since it seemed like the least time wasted overall (open to feedback here, though). The cht-datasource code that is using `fetch` is structured to be reusable so references to the actual `fetch` function should be isolated and easy to update if necessary in the future.

Reply:
Another option is to have an optional injection for fetch so API can inject [node-fetch](https://www.npmjs.com/package/node-fetch) which should be a drop in replacement.

Let's go with `fetch` for now and rely on e2e testing to find any issues in the Node 20 implementation or when we upgrade to Node 22.